back/engines/free/polls/app/services/polls/xlsx_service.rb

```diff
@@ -11,11 +11,43 @@ def generate_poll_results_xlsx participation_context, responses
       columns = if is_anonymous
         []
       else
-        [{
-          header: 'User ID', 
-          f: -> (r) { r.user_id },
-          skip_sanitization: true
-        }]
+				if current_user.admin?
+	        [
+						{
+		          header: 'User ID',
+		          f: -> (r) { r.user_id },
+		          skip_sanitization: true
+		        },
+						{
+		          header: 'First Name',
+		          f: -> (r) { r.user.first_name },
+		        },
+						{
+		          header: 'Last Name',
+		          f: -> (r) { r.user.first_name },
+		        },
```

Comment:
Some duplication here that we can easily avoid. I'd suggest to put this in a private helper method. Something along those lines (didn't test it):
```ruby
def compute_user_columns(is_anonymous, is_admin)
return [] if is_anonymous

columns = [
	{ header: 'User ID',    f: -> (r) { r.user_id }, skip_sanitization: true },
	{ header: 'First Name', f: -> (r) { r.user.first_name } },
	{ header: 'Last Name',  f: -> (r) { r.user.first_name } }
]

columns << { header: 'Email', f: -> (r) { r.user.email }} if is_admin
columns
end
```